File: Top K Frequent Elements/Top_K_Frequent_Elements.cpp
class Solution {
public:
    static bool compare(pair<int,int> &t1,pair<int,int> &t2) {
        if(t1.second > t2.second) return true;
        return false;
    }
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> mpp;
        for(auto i: nums) {
            mpp[i]++;
        }
        vector<pair<int,int>> temp;
        for(auto i: mpp) {
            temp.push_back({i.first,i.second});
        }
        sort(temp.begin(),temp.end(),compare);
        vector<int> ans;
        for(int i=0;i<k; i++) {
            ans.push_back(temp[i].first);
        }
        return ans;
    }
};

File: Top K Frequent Elements/Top_K_Frequent_Elements.java
class Solution {

    public static int compare(Map.Entry<Integer, Integer> t1, Map.Entry<Integer, Integer> t2) {
        return t2.getValue() - t1.getValue();
    }

    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> mpp = new HashMap<>();
        for (int i : nums) {
            mpp.put(i, mpp.getOrDefault(i, 0) + 1);
        }
        
        List<Map.Entry<Integer, Integer>> temp = new ArrayList<>(mpp.entrySet());
        temp.sort(Solution::compare);
        
        int[] ans = new int[k];
        for (int i = 0; i < k; i++) {
            ans[i] = temp.get(i).getKey();
        }
        return ans;
        
    }
}